client/shared/interfaces/user_records_interface.coffee

```diff
@@ -22,6 +22,9 @@ module.exports = class UserRecordsInterface extends BaseRecordsInterface
   deactivate: (user) =>
     @remote.post 'deactivate', user.serialize()
 
+  destroy: (user) =>
```

Comment:
Seems like you don't need the user to destroy here, meaning we can avoid the slightly awkward
```
Records.users.destroy(Session.user()) # could just be Records.users.destroy()
```